internal/verifier/sigstore/sigstore.go

```diff
@@ -70,12 +70,11 @@ func New(trustedRoot string, authOpts ...container.AuthMethod) (*Sigstore, error
 	}
 
 	// init sigstore's verifier
-	trustedrootJSON, err := tuf.GetTrustedrootJSON(trustedRoot, cacheDir)
-	if err != nil {
-		return nil, err
-	}
+	tufOpts := tuf.DefaultOptions()
+	tufOpts.Root = []byte(trustedRoot)
```

Comment:
I don't think this works (but at the same time the variable is misnamed and our tests don't catch this).

our trusted root is just the name of the sigstore instance (e.g. `"tuf-repo-cdn.sigstore.dev`). If you want to pass the TUF root bytes, we can instead just pass in what `embeddedRootJson` returns or set `RepositoryBaseURL` to a URL of the sigstore instance, maybe defaulting to `https:// + trustedRoot`

Reply:
OK, I factored out the root init and added tests. Also, I added a `readRootJson()` utility func to handle looking up cached root data (to pair it with the existing `writeRootJson`).

PTAL @jhrozek  :)